### PR TITLE
Update IB documentation and misc enhancements / bug fixes

### PR DESCRIPTION
**Pull request by benjaminsingleton**
# Pull Request

- Rewrote all of the Interactive Brokers adapter documentation. I wrote the docs the way I wish they had been written when I first started exploring NT. Hope others find it helpful.
- Updated docker image for IB Gateway and other fixes. Existing image has been abandoned for 6+ months and couldn't get it working.
- Added `host` and `port` to `InteractiveBrokersGatewayConfig`
- Added `contract_details_to_ib_contract_details` helper function
- Other minor enhancements and bug/documentation fixes

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## How has this change been tested?

- [X] Test InteractiveBrokersGateway works properly

### CODE REVIEW

**Comment by cjdsellers** [comment]
Hey @benjaminsingleton 

Thanks so much! I'll wait until you mark this as ready for review before merging 🙏

**Review by rsmb7z — APPROVED** [review]

**Comment by cjdsellers** [comment]
Thanks for the review @rsmb7z will merge now! 👏

**Pull request merged**
